src/core/api/chart_repository.go

```diff
@@ -549,3 +576,44 @@ func chartFullName(namespace, chartName, version string) string {
 	}
 	return fmt.Sprintf("%s/%s:%s", namespace, chartName, version)
 }
+
+func (cra *ChartRepositoryAPI) getFilename() (string, error) {
+	r := cra.Ctx.Request
+	err := r.ParseForm()
+	if err != nil {
+		hlog.Errorf("failed to parse chart form: %v", err)
+		return "", err
+	}
+	filename := r.PostFormValue("filename")
+	if filename == "" {
+		return "", errors.New("empty chart filename")
+	}
+	return filename, nil
+}
+
+// getChartNameAndVersion: filename format  demo-0.1.0.tgz
+func getChartNameAndVersion(filename string) (string, string) {
```

Comment:
filename is not actually a naming convention of chart tgz, you cannot think it always with a formatted name.

Reply:
> filename is not actually a naming convention of chart tgz, you cannot think it always with a formatted name.

chart event related notification code deleted